plugins/inputs/prometheus/kubernetes.go

```diff
@@ -180,14 +367,21 @@ func registerPod(pod *corev1.Pod, p *Prometheus) {
 		return
 	}
 	podURL := p.AddressToURL(URL, URL.Hostname())
-	p.lock.Lock()
+
+	// Locks earlier if using cAdvisor calls - makes a new list each time
+	// rather than updating and removing from the same list
+	if p.MonitorPodsVersion != 2 {
+		p.lock.Lock()
```

Comment:
if you use defer you can drop the second conditional.
```
		p.lock.Lock()
    defer p.lock.Unlock()
```
But you may want to move the locks out of here as the caller should probably lock.

Reply:
I have added the defer to drop the second conditional